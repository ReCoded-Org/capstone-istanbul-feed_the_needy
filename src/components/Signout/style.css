.logoutButton {
  background-color: unset;
  border: unset;
}

.logoutButton:hover {
  cursor: pointer;
}
