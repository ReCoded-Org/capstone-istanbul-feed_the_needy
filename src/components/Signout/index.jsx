import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Modal, message } from "antd";
import "./style.css";
import { useTranslation } from "react-i18next";
import firebase from "../../firebaseConfig";

const auth = firebase.auth();

const Signout = ({ setUserExist }) => {
  const { t } = useTranslation();
  const [modalVisible, setModalVisible] = useState(false);
  const history = useHistory();

  const handleOk = () => {
    auth
      .signOut()
      .then(() => {
        setModalVisible(false);
        setUserExist(false);
        message.success("Successfully logged out!");
      })
      .catch((error) => {
        message.warning(error.message);
      });
    history.push("/");
  };

  return (
    <>
      <button
        className="logoutButton"
        type="button"
        onClick={() => setModalVisible(true)}
      >
        {t("navbar.headers.header3")}
      </button>
      <Modal
        title={t("navbar.headers.header3")}
        visible={modalVisible}
        onOk={handleOk}
        onCancel={() => setModalVisible(false)}
        locale={{ okText: t("logout.ok"), cancelText: t("logout.cancel") }}
      >
        <p>{t("logout.text")}</p>
      </Modal>
    </>
  );
};

export default Signout;
