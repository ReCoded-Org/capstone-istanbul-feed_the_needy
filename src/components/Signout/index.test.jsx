import React from "react";
import renderer from "react-test-renderer";
import Signout from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("Signout renders correctly", () => {
  const tree = renderer.create(<Signout />).toJSON();
  expect(tree).toMatchSnapshot();
});
