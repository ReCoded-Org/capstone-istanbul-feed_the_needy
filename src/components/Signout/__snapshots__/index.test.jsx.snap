// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signout renders correctly 1`] = `
<button
  className="logoutButton"
  onClick={[Function]}
  type="button"
>
  navbar.headers.header3
</button>
`;
